STIPS/STIP-005.md

```diff
@@ -0,0 +1,1239 @@
++ *usdcAmountIn = (vBaseCostReal / current leverage) + slippageCost + owedRealizedPnLDiscount*
++ *setToken.externalPositionUnit = usdcAmountIn / mintQuantity*
+
+</td></tr>
+<tr><td valign="top">componentIssuanceHook</td>
+<td>  
+
+Invoked as an externalPositionHook duuring DIM.resolveEquityPositions. Happens after DIM has read external USDC position (set during moduleIssuanceHook) and transferred in:
++ *usdcAmount = mintQuantity * setToken.externalPositionUnit*
+
+Deposit USDC from SetToken into PerpV2
++ *SetToken.invokeApprove(USDC, PerpV2Vault, usdcAmount)*  
++ *SetToken.invokeDeposit(PerpV2Vault, USDC, usdcAmount)*
+
+Open position  
+```solidity
+swapParams = {
+  baseToken: vBaseToken 
+  isBaseToQuote: false		// long
+  isExactInput: true		
+  amount: usdcAmount
+  sqrtPriceLimitX96: 0		// slippage protection
+}
+```
+  
++ *PerpV2ClearingHouse.openPosition(swapParams)*
+  
+</td>
+</tr>
+</table>
+
+
+
+## Redemption
+
+This section includes:
++ A flow chart and user story for generalized PerpModule with protocol specific logic delegated to a PerpAdapter
++ A detailed walkthrough of redemption logic that would be implemented for the PerpV2 protocol
+
+![](../assets/stip-005/perp_userflow_redeem.png "")
+
+### User Story: Redeeming 10 Sets 
+
+UserA owns 10 SetTokens which represent a leveraged Perp position in ETH and wants to redeem them for the Perp’s collateral token, USDC.  
+
+1. User calls *DIM.redeem(setToken, 10)*
+
+2. DIM calls *PerpModule moduleRedeemHook(setToken, 10)* which passes call through to PerpAdapter contract for protocol 
+   specific implementation.
+
+3. PerpAdapter:
+     + Reads vETH position size, margin debt, and collateral positions from protocol, calculating their position units
+     + Calculates amount of pending payments (funding, carried PnL) which existing set holders must settle as a position unit
+
+4. PerpAdapter calls PerpProtocol to sell *vETHPositionUnit * 10* amount (see Trade user flow for details). 
+   Using the quote delta amount generated by the swap, it calculates amount of USDC to return for redemption, making sure 
+   redeemer pays slippage and their share of pending funding fees, carried PnL
+      + *realizedPnL = deltaQuote - | expected debt reduction |*
+      + *usdcOut = (collateral per set * 10) - (funding debt per set * 10) + realizedPnL*
+
+5. PerpModule sets the SetToken *externalPositionUnit* to: *usdcOut / 10*
+
+6. After burning 10 SetTokens, DIM reads SetToken’s *externalPositionUnit* and calculates the usdc *componentQuantity* to 
+   return to user
+
+7. DIM resolves the equity position by calling PerpModule’s *externalPositionHook* which calls the PerpAdapter’s redeem hook 
+   implementation  
+
+8. PerpAdapter calls *PerpProtocol.withdraw(setToken, componentQuantity)*
+
+9. PerpProtocol calls *USDC.transfer(setToken, componentQuantity)* converting correct amount of  SetToken’s USDC external position into a USDC default position. 
```

Comment:
We aren't updating the USDC Default position, so let's be clear and just say the USDC is being transferred back to the SetToken